File: Python/CaesarCipher/CrackingBruteForce.py
ALPHABET = ' ABCDEFGHIJKLMNOPQRSTUVWXYZ'
KEY = 0

def crack_caesar(cipher_text):
    for key in range(len(ALPHABET)):
        plain_text = ''
        for c in cipher_text:
            index = ALPHABET.find(c)
            index = (index - key) % len(ALPHABET)
            plain_text = plain_text + ALPHABET[index]
        print('with key %s, the result is: %s' % (key, plain_text))

if __name__ == '__main__':
    encrypted_text = 'LCDPCWKHCFLSKHUCPHVVDJH'
    crack_caesar(encrypted_text)

File: Java/CaesarCipher/CrackingBruteForce.java
package Java.CaesarCipher;

public class CrackingBruteForce {

    private String ALPHABET = " ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public void crack(String cipherText) {

        for (int key = 0; key < ALPHABET.length(); key++) {
            String plainText = "";
            for (int i = 0; i < cipherText.length(); i++) {
                char character = cipherText.charAt(i);
                int charIndex = ALPHABET.indexOf(character);
                // int decryptedIndex = (charIndex - key) % ALPHABET.length();
                int decryptedIndex = Math.floorMod(charIndex - key, ALPHABET.length());
                plainText += ALPHABET.charAt(decryptedIndex);
            }
            System.out.format("Cracking Caesar-Cipher with %s key the result is: %s%n",
                 key, plainText);
        }
    }
}
